filter/ip2proxy/filterip2proxy.go

```diff
@@ -0,0 +1,206 @@
+	QuietFail   bool     `json:"quiet" yaml:"quiet"`               // fail quietly
+	SkipPrivate bool     `json:"skip_private" yaml:"skip_private"` // skip private IP addresses
+	PrivateNet  []string `json:"private_net" yaml:"private_net"`   // list of own defined private IP addresses
+	CacheSize   int      `json:"cache_size" yaml:"cache_size"`     // cache size
+
+	db    *ip2proxy.DB
+	dbMtx sync.RWMutex
+
+	cache        *lru.Cache
+	privateCIDRs []*net.IPNet
+
+	watcher *fsnotify.Watcher
+	ctx     context.Context
+}
+
+// DefaultFilterConfig returns an FilterConfig struct with default values
+func DefaultFilterConfig() FilterConfig {
+	return FilterConfig{
+		FilterConfig: config.FilterConfig{
+			CommonConfig: config.CommonConfig{
+				Type: ModuleName,
+			},
+		},
+		Key:         ModuleName,
+		QuietFail:   false, // backwards compatible
+		SkipPrivate: false,
+		CacheSize:   100000,
+	}
+}
+
+// InitHandler initialize the filter plugin
+func InitHandler(ctx context.Context, raw *config.ConfigRaw) (config.TypeFilterConfig, error) {
+	conf := DefaultFilterConfig()
+	err := config.ReflectConfig(raw, &conf)
+	if err != nil {
+		return nil, err
+	}
+
+	conf.db, err = ip2proxy.OpenDB(conf.DBPath)
+	if err != nil {
+		return nil, err
+	}
+
+	conf.cache, err = lru.New(conf.CacheSize)
+	if err != nil {
+		return nil, err
+	}
+
+	conf.ctx = ctx
+
+	var cidrs []string
+	if len(conf.PrivateNet) > 0 {
+		cidrs = conf.PrivateNet
+	} else {
+		cidrs = ip2l.DefaultCIDR
+	}
+
+	// init fsnotify
+	goglog.Logger.Infof("%s fsnotify initialized for %s", ModuleName, conf.DBPath)
+	conf.watcher, err = fsnotify.NewWatcher()
+	if err != nil {
+		goglog.Logger.Errorf("%s failed to init watcher, %s", ModuleName, err.Error())
+	}
+	err = conf.watcher.Add(conf.DBPath)
+	if err != nil {
+		goglog.Logger.Errorf("%s failed to add file: %s", ModuleName, err.Error())
+	}
+	conf.initFsnotifyEventHandler()
+
+	for _, cidr := range cidrs {
+		_, privateCIDR, err := net.ParseCIDR(cidr)
+		if err != nil {
+			return nil, err
+		}
+		conf.privateCIDRs = append(conf.privateCIDRs, privateCIDR)
+	}
+
+	return &conf, nil
+}
+
+// not_supported is copied from ip2proxy and is the field entry for each field that is not supported by the current database
+const not_supported string = "NOT SUPPORTED"
+
+// Event the main filter event
+func (f *FilterConfig) Event(ctx context.Context, event logevent.LogEvent) (logevent.LogEvent, bool) {
+	ipstr := event.GetString(f.IPField)
+	if ipstr == "" {
+		// Passthru if empty
+		return event, false
+	}
+	ip := net.ParseIP(ipstr)
+	if ip == nil {
+		return event, false
+	}
+	if f.SkipPrivate && f.privateIP(ip) {
+		// Passthru
+		return event, false
+	}
+	var record map[string]string
+	// single-thread here
+	if c, ok := f.cache.Get(ipstr); ok {
+		record = c.(map[string]string)
+	} else {
+		var err error
+		f.dbMtx.RLock()
+		record, err = f.db.GetAll(ipstr)
+		f.dbMtx.RUnlock()
+		if err != nil {
+			if f.QuietFail {
```

Comment:
shouldn't the logic be reversed?

Reply:
Hi

This is also code that is copied from filtergeoip2. (And also used in filterip2location.)

I agree that it looks like it should be reversed based on the code and the comments.

How do you want to proceed? I suggest you to accept this PR, and I will then create a separate PR where I revert the logic and add the appropriate documentation for all three filters.

Reply:
ok